File: ruby/udemy_class/stacks_deques_and_queues/queue.rb
class Queue
  def initialize(store = [])
    @store = store
  end

  def enqueue(value)
    @store.unshift(value)
  end

  def dequeue
    @store.pop
  end

  def size
    @store.length
  end

  def is_empty
    size == 0
  end

  def inspect
    @store
  end
end

q = Queue.new
q.enqueue(1)
q.enqueue(2)
print q.inspect
q.dequeue
print q.inspect

File: python/udemy_class/stacks_deques_and_queues/queue.py
class Queue:
    def __init__(self, store = []):
        self.store = store

    def enqueue(self, value):
        self.store.insert(0, value)

    def dequeue(self):
        self.store.pop()

    def size(self):
        return len(self.store)

    def is_empty(self):
        return self.size() == 0

    def inspect(self):
        return self.store

q = Queue()
q.enqueue(1)
q.enqueue(2)
print(q.inspect())
q.dequeue()
print(q.inspect())
